core/src/main/java/org/apache/cxf/attachment/AttachmentSerializer.java

```diff
@@ -327,7 +327,7 @@ public void writeAttachments() throws IOException {
         StringWriter writer = new StringWriter();
         writer.write("\r\n--");
         writer.write(bodyBoundary);
-        writer.write("--");
+        writer.write("--\r\n");
```

Comment:
@gchauvet this is not correct as per rfc1341 (see please https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html):

```
Overall, the body of a multipart entity may be specified as follows:
multipart-body := preamble 1*encapsulation 
               close-delimiter epilogue 

encapsulation := delimiter CRLF body-part 

delimiter := CRLF "--" boundary   ; taken from  Content-Type 
field. 
                               ;   when   content-type    is 
multipart 
                             ; There must be no space 
                             ; between "--" and boundary. 

close-delimiter := delimiter "--" ; Again, no  space  before 
"--" 

preamble :=  *text                  ;  to  be  ignored  upon 
receipt. 

epilogue :=  *text                  ;  to  be  ignored  upon 
receipt. 

body-part = <"message" as defined in RFC 822, 
         with all header fields optional, and with the 
         specified delimiter not occurring anywhere in 
         the message body, either on a line by itself 
         or as a substring anywhere.  Note that the 
         semantics of a part differ from the semantics 
         of a message, as described in the text.>
```


The element in question is 
```
close-delimiter := delimiter "--" ; Again, no  space  before 
```
"--"

Reply:
hello @reta, 

Unless I've misinterpreted the quote below, aren't we supposed to have a CRLF?
> Note that the encapsulation boundary must occur at the beginning of a line, i.e., following a CRLF, and that that initial CRLF is considered to be part of the encapsulation boundary rather than part of the preceding part. The boundary must be followed immediately either by another CRLF and the header fields for the next part, or by two CRLFs, in which case there are no header fields for the next part (and it is therefore assumed to be of Content-Type text/plain).

Or the webservice I'm calling follows RFC 2046 (https://www.rfc-editor.org/rfc/rfc2046#page-44):
```
multipart-body := [preamble CRLF]
                  dash-boundary transport-padding CRLF
                  body-part *encapsulation
                  close-delimiter transport-padding
                  [CRLF epilogue]

Reply:
Okay, this is a trick from the webservice provider for security purpose. We've found a workaround, there's no longer any problem. thank you.